File: AgregacionComposicion/Ejercicio1/Habitacion.java
// a) Implementa las clases con sus constructores, getters y setters.
class Habitacion {
    private String nombre;
    private double tamano;

    public Habitacion(String nombre, double tamano) {
        this.nombre = nombre;
        this.tamano = tamano;
    }

    public String getNombre() {
        return nombre;
    }

    public double getTamano() {
        return tamano;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public void setTamano(double tamano) {
        this.tamano = tamano;
    }

    public void mostrarInfo() {
        System.out.println("Habitación: " + nombre + ", Tamaño: " + tamano + " m²");
    }
}

import java.util.ArrayList;
import java.util.List;

class Casa {
    private String direccion;
    private List<Habitacion> habitaciones;

    public Casa(String direccion) {
        this.direccion = direccion;
        this.habitaciones = new ArrayList<>();
    }

    public String getDireccion() {
        return direccion;
    }

    public List<Habitacion> getHabitaciones() {
        return habitaciones;
    }

    public void setDireccion(String direccion) {
        this.direccion = direccion;
    }

    public void agregarHabitacion(Habitacion habitacion) {
        habitaciones.add(habitacion);
    }

    public void mostrarCasa() {
        System.out.println("Dirección: " + direccion);
        for (Habitacion h : habitaciones) {
            h.mostrarInfo();
        }
    }
}

public class Main {
    public static void main(String[] args) {
        // b) Crea una casa y agrega varias habitaciones.
        Casa miCasa = new Casa("Av. Siempre Viva 742");
        miCasa.agregarHabitacion(new Habitacion("Sala", 25.0));
        miCasa.agregarHabitacion(new Habitacion("Cocina", 15.5));
        miCasa.agregarHabitacion(new Habitacion("Dormitorio", 20.0));

        // c) Muestra la información de la casa y sus habitaciones.
        miCasa.mostrarCasa();
    }
}

File: AgregacionComposicion/Ejercicio1/Habitacion.py
# a) Implementa las clases con sus constructores, getters y setters.
class Habitacion:
    def __init__(self, nombre, tamano):
        self.nombre = nombre
        self.tamano = tamano

    def mostrar_info(self):
        print(f'Habitación: {self.nombre}, Tamaño: {self.tamano} m²')


class Casa:
    def __init__(self, direccion):
        self.direccion = direccion
        self.habitaciones = []

    def agregar_habitacion(self, habitacion):
        self.habitaciones.append(habitacion)

    def mostrar_casa(self):
        print(f'Dirección: {self.direccion}')
        for h in self.habitaciones:
            h.mostrar_info()


# b) Crea una casa y agrega varias habitaciones.
mi_casa = Casa("Av. Siempre Viva 742")
mi_casa.agregar_habitacion(Habitacion("Sala", 25.0))
mi_casa.agregar_habitacion(Habitacion("Cocina", 15.5))
mi_casa.agregar_habitacion(Habitacion("Dormitorio", 20.0))

# c) Muestra la información de la casa y sus habitaciones.
mi_casa.mostrar_casa()
